### ISSUE TITLE
Custom type on the basis  map[string]interface{}, error marshaling

### ISSUE THREAD

**Issue by gebv** [vmihailenco/msgpack#90]
I have error when using custom type on the basis map[string]interface{}
How to use?

Error:
```
panic: interface conversion: interface {} is store.S, not map[string]interface {}
```

Ex. code:
```
package store

import (
	"testing"
)

type S map[string]interface{}

func (s S) Set(key string, v interface{}) {
	s[key] = v
}

func (s S) Get(key string) interface{} {
	return s[key]
}

func TestStore(t *testing.T) {
	msgpack.RegisterExt(1, S{})
	
	v := S{}
	
	v.Set("a", "a")
	assert.Equal(t, v.Get("a"), "a")
	
	network, err := msgpack.Marshal(v)
	assert.Equal(t, err, nil, "encode")
	
	// 
	
	v = S{}
	err = msgpack.Unmarshal(network, &v)
	assert.Equal(t, err, nil, "decode")
	
	assert.Equal(t, v.Get("a"), "a", "custom type")
}
```

**Comment by gebv** [comment]
gopkg.in/vmihailenco/msgpack.v2/encode_map.go:66

Why not use a registered type?

**Comment by vmihailenco** [comment]
>Why not used a registered type?

Then I can't apply optimization to your `type S map[string]interface{}`, because `S` and `map[string]interface{}` have different types.

**Comment by vmihailenco** [comment]
Fixed in v2.5.4. Thanks for the report.

**Comment by gebv** [comment]
I appreciate the urgency
